pkg/kn/commands/broker/delete.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"knative.dev/client/pkg/kn/commands"
+)
+
+var delete_example = `
+# Delete a broker 'mybroker' in the current namespace
+  kn broker create mybroker
+# Delete a broker 'mybroker' in the 'myproject' namespace
+  kn broker create mybroker --namespace myproject
+`
+
+func NewBrokerDeleteCommand(p *commands.KnParams) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:     "delete NAME",
+		Short:   "Delete a broker.",
+		Example: delete_example,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if len(args) != 1 {
+				return errors.New("'broker delete' requires the broker name given as single argument")
+			}
+			name := args[0]
```

Comment:
Here I'm less sure whether we should default to "default", as this could lead to an accidental deletion of a broker.

It really depends how likely the use case to have multiple brokers in a namespace is. If 95% of all usescases only use the default broker, then having a default value is helpful as this supports the user (who might not even now that he can have more than one broker).